legend-pure-code-compiled-core-relational/src/main/resources/core_relational/relational/sqlQueryToString/tests/testSpannerCompatibility.pure

```diff
@@ -0,0 +1,351 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+###Pure
```

Comment:
Can we try clubbing the tests for diff dbType. Maybe somehow parameterizing it on dbType, so that it's easier next time?
Just a though

Reply:
I agree in principle though the devil is in the details. Atleast with BigQuery and Spanner there is enough difference in behavior that we cannot just have a "simple" set of tests that are parameterized by the db type. We will have to support a notion of also parameterizing not only the db type but also the behavior. For e.g for BigQuery sql should use "substr" vs for Spanner it should use "substring"

Reply:
Actually I meant we have common set of pure queries but different asserts for diff db.
So that next time we don't write pure queries again just write the assert strings
But maybe we can do it in another MR